app/helpers/works_helper.rb

```diff
@@ -96,12 +96,16 @@ def marked_for_later?(work)
     reading && reading.toread?
   end
 
+  def markasread_link(work)
+    link_to ts("Mark as Read"), marktoread_work_path(work)
+  end
+
   def marktoread_link(work)
     link_to ts("Mark for Later"), marktoread_work_path(work)
   end
 
-  def markasread_link(work)
-    link_to ts("Mark as Read"), marktoread_work_path(work)
+  def removemarktoread_link(work)
```

Comment:
Because marking as read should remove the item from your marked for later list. Isn't this just adding it if it uses marktoread_work_path(work)?